viewer/src/views/threejs/cad/CadNode.ts

```diff
@@ -81,11 +81,39 @@ export class CadNode extends THREE.Object3D {
     this.renderMode = RenderMode.Color;
   }
 
+  setColor(treeIndex: number, red: number, green: number, blue: number) {
+    this._materials.colorDataTexture.image.data[4 * treeIndex] = red;
+    this._materials.colorDataTexture.image.data[4 * treeIndex + 1] = green;
+    this._materials.colorDataTexture.image.data[4 * treeIndex + 2] = blue;
+    this._materials.colorDataTexture.image.data[4 * treeIndex + 3] = 255;
+    this._materials.colorDataTexture.needsUpdate = true;
+  }
+
+  resetColor(treeIndex: number) {
+    this._materials.colorDataTexture.image.data[4 * treeIndex] = 0;
+    this._materials.colorDataTexture.image.data[4 * treeIndex + 1] = 0;
+    this._materials.colorDataTexture.image.data[4 * treeIndex + 2] = 0;
+    this._materials.colorDataTexture.image.data[4 * treeIndex + 3] = 0;
+    this._materials.colorDataTexture.needsUpdate = true;
+  }
+
   set renderMode(mode: RenderMode) {
     this._renderMode = mode;
-    for (const material of Object.values(this._materials)) {
-      material.uniforms.renderMode.value = mode;
-    }
```

Comment:
Because the `Materials` object now contains both materials and a texture. Iterating in this way is not very type-safe, because TypeScript doesn't care to check if all `Object.values` has proper types. So this loop failed spectacularly at runtime instead.